STYLE.md

```diff
@@ -109,6 +109,8 @@ A few general notes on our error handling philosophy:
   that there is complete test coverage for the degraded case.** Additionally, the user should be
   aware of the degraded state minimally via an error log of level warn or greater and via the
   increment of a stat.
+* If you do need to log a non-fatal warning or error, you can unit-test it by instantiating a 
+  MockLogSink from test/mocks/common.h.
```

Comment:
Can we add some statement along the lines of "It's generally bad practice to test by depending on log messages unless the actual behavior being validated is logging. It's preferable to export statistics to enable consumption by external monitoring for any behavior that should be externally consumed or to introduce appropriate internal interfaces such as mocks for internal behavior.".

I'm somewhat worried about it becoming too easy to  depend on logging in tests now that we have a great facility for it ;)

Reply:
done